tests/framework/rest/UrlRuleTest.php

```diff
@@ -340,19 +383,23 @@ public function createUrlDataProvider()
      */
     public function testCreateUrl($rule, $tests)
     {
+        $this->mockWebApplication();
+        Yii::$app->set('request', new Request([
+            'hostInfo' => 'http://api.example.com',
+            'scriptUrl' => '/index.php'
+        ]));
+        $manager = new UrlManager([
+            'cache' => null,
+        ]);
+        $rule = new UrlRule($rule);
```

Comment:
`$rule = new UrlRule($rule);` is dangerous and will be changed #14998 

and I like this
realization out of foreach is better for me, I do it same